lib/Doctrine/ORM/EntityManager.php

```diff
@@ -191,6 +200,16 @@ public function getMetadataFactory()
     }
 
     /**
+     * Gets the persister factory used to manipulate collection and entity classes.
+     *
+     * @return \Doctrine\ORM\Persisters\PersisterFactory
+     */
+    public function getPersisterFactory()
```

Comment:
Hibernate exposes it at EntityManager level because it can be swappable with another implementation.
Default implementation is also reachable from outside, allowing you to provide a ServiceInjector (read as DI Container).
I'm ok to move this to UnitOfWork, but I would not consider this as an extension point (the part I wanted to discuss internally). This means there'll be no interface bound to PersisterFactory.
Right now I made it part of EntityManager and wanted to discuss IF we'd make it an extension point.